blocks/block-edit/index.js

```diff
@@ -19,8 +19,18 @@ import {
 	getBlockDefaultClassName,
 	hasBlockSupport,
 } from '../api';
+import { BlockEditContextProvider } from './context';
 
 export class BlockEdit extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
```

Comment:
I don't think a component needs _both_ the initial setting of state _and_ `static getDerivedStateFromProps`. The latter supersedes the need for the former. The constructor could be dropped altogether.

https://codepen.io/aduth/pen/QmRZLO

**Edit:** That said, it appears to be needed if you plan to `return null;` from `static getDerivedStateFromProps`, since otherwise the initial value of `this.state` is `undefined`.

https://codepen.io/aduth/pen/NYVOPa

Reply:
Since `getDerivedStateFromProps` is _always_ called on the initial mount, do we really need to set the values of state in the constructor? Or can we just set it to an empty object?

Reply:
Yes, let's do that. We will probably have to use `get` to check state, but in overall it's still better.

Reply:
Much better: https://github.com/WordPress/gutenberg/pull/5029/commits/fbb66c8052514e13b077ccaa51f09541d934e8f9